livespark-form-modeler/livespark-form-modeler-codegen/livespark-form-modeler-codegen-impl/src/test/java/org/livespark/formmodeler/codegen/flow/impl/RoasterFlowProducerSourceGeneratorTest.java

```diff
@@ -0,0 +1,188 @@
+
+        context = new SourceGenerationContext(formDefinition, path, root, local, shared, null, projectForms);
+        generatedSource = sourceGenerator.generateJavaSource(context);
+        gslen = generatedSource.length();
+    }
+
+    @Test
+    public void generateJavaSource() {
+        assertThat(getImports()).containsAll(IMPORTS);
+
+        assertThat(getPackage()).isEqualTo(PACKAGE);
+        assertThat(getClassAnnotation()).isEqualTo(CLASS_ANNOTATION);
+        assertThat(getFullClassName()).isEqualTo(FULL_CLASS_NAME);
+
+        assertThat(getAbstractMethodsImplementations()).isEqualTo(ABSTRACT_METHODS_IMPLEMENTATIONS);
+        assertThat(getProducerMethodsImplementations()).isEqualTo(PRODUCER_METHODS_IMPLEMENTATIONS);
+
+        Map<String, List<String>> methodsAnnotations = getAllMethodsAnnotations();
+        for (String abstractMethod : ABSTRACT_METHODS) {
+            assertThat(methodsAnnotations.get(abstractMethod)).isEqualTo(Arrays.asList("Override"));
+        }
+        for (String producerMethod : PRODUCER_METHODS.subList(1, 5)) {
+            assertThat(methodsAnnotations.get(producerMethod)).containsExactlyElementsOf(Arrays.asList("Override", "Produces", "Singleton", "ForEntity(\"org.juraj.mypackage.myFormName\")", "Named(\"" + producerMethod + "\")"));
+        }
+    }
+
+    private Set<String> getImports() {
+        Set<String> imports = new HashSet<>();
+        int start = generatedSource.indexOf("import") + 6;
+        if (start != 5) {
+            int end;
+            char nextChar;
+            StringBuilder nextImport;
+            do {
+                nextImport = new StringBuilder();
+                end = start + 1;
+                do {
+                    nextChar = generatedSource.charAt(end++);
+                    if (nextChar != ';') {
+                        nextImport.append(nextChar);
+                    } else {
+                        break;
+                    }
+                } while (true);
+                imports.add(nextImport.toString());
+                start = end + 7;
+            } while (generatedSource.charAt(end + 1) == 'i');
+        }
+        return imports;
+    }
+
+    private String getPackage() {
+        return getStringThatEndsWithChar(PACKAGE_INDEX, ';');
+    }
+
+    private String getClassAnnotation() {
+        return getStringThatEndsWithChar(generatedSource.indexOf("@"), '\n');
+    }
+
+    private String getFullClassName() {
+        return getStringThatEndsWithChar(generatedSource.indexOf("public"), '\n');
+    }
+
+    private List<String> getAbstractMethodsImplementations() {
+        return ABSTRACT_METHODS.stream().map(this::getMethodImplementation).collect(Collectors.toList());
+    }
+
+    private List<String> getProducerMethodsImplementations() {
+        return PRODUCER_METHODS.stream().map(this::getMethodImplementation).collect(Collectors.toList());
+    }
```

Comment:
Nice use of streams! However, both methods now have the exact same code, and could in fact be refactored into one method `getMethodImplementations` with an argument.